cluster_management/src/main/java/com/pinterest/rocksplicator/BulkIngestionStateModelFactory.java

```diff
@@ -0,0 +1,150 @@
+/// Copyright 2017 Pinterest Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+/// http://www.apache.org/licenses/LICENSE-2.0
+
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+//
+// @author cdonaghy (cdonaghy@pinterest.com)
+//
+
+package com.pinterest.rocksplicator;
+
+import com.pinterest.rocksdb_admin.thrift.CheckDBResponse;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.apache.helix.HelixAdmin;
+import org.apache.helix.NotificationContext;
+import org.apache.helix.model.ExternalView;
+import org.apache.helix.model.Message;
+import org.apache.helix.participant.statemachine.StateModel;
+import org.apache.helix.participant.statemachine.StateModelFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+public class BulkIngestionStateModelFactory extends StateModelFactory<StateModel> {
+  private final String host;
+  private final int adminPort;
+  private final String cluster;
+  private CuratorFramework zkClient;
+
+  private static final Logger LOG = LoggerFactory.getLogger(BulkIngestionStateModelFactory.class);
+
+  BulkIngestionStateModelFactory(String host, int adminPort, String zkConnectString, String cluster) {
+    this.host = host;
+    this.adminPort = adminPort;
+    this.cluster = cluster;
+    this.zkClient = CuratorFrameworkFactory.newClient(zkConnectString,
+        new ExponentialBackoffRetry(1000, 3));
+    zkClient.start();
+  }
+
+  @Override
+  public StateModel createNewStateModel(String resourceName, String partitionName) {
+    LOG.error("Create a new state for " + partitionName);
+    return new BulkIngestionStateModel(resourceName, partitionName, adminPort, cluster, zkClient, host);
+  }
+
+  public static class BulkIngestionStateModel extends OnlineOfflineStateModelFactory.OnlineOfflineStateModel {
```

Comment:
How do you plan to setup the replication graph if it extends from OnlineOfflineStateModel.

It would be nice if you could summarize the idea about how do you plan to support polaris use case.